mail-server/monit.nix

```diff
@@ -0,0 +1,38 @@
+#  nixos-mailserver: a simple mail server
+#  Copyright (C) 2016-2017  Robin Raymond
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program. If not, see <http://www.gnu.org/licenses/>
+
+{ config, pkgs, lib, ... }:
+
+let
+  cfg = config.mailserver;
+in
+{
+  config = lib.mkIf cfg.monitoring {
+    services.monit = {
+      enable = true;
+      config = ''
+        set daemon 120
+        set alert root@localhost
+        set httpd port 2812 and use address localhost
+            allow localhost
+            allow admin:obwjoawijerfoijsiwfj29jf2f2jd
```

Comment:
what is this?

Reply:
this is what I believe to be the minimal config monit requires to work.
The funny looking line at the end is the password for the user "admin" using HTTP Basic Auth.
In the end I doubt it matters much what is written there since the server is not reachable from outside of localhost. But it doesn't hurt to have some strong password there because the monit tools themselves don't require you to input the password (e.g. `monit report`).